controllers/provisioning_controller.go

```diff
@@ -222,49 +218,22 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, errors.Wrap(err, "failed to create Inspector password")
 	}
 
-	// If Metal3 Deployment already exists, do nothing.
-	exists, err := provisioning.GetExistingMetal3Deployment(r.KubeClient.AppsV1(), ComponentNamespace)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return ctrl.Result{}, errors.Wrap(err, "failed to find existing Metal3 Deployment")
-	}
-
-	if exists {
-		r.Log.V(1).Info("metal3 deployment already exists")
-		err = r.updateCOStatus(ReasonComplete, "found existing Metal3 deployment", "")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Available state: %v", clusterOperatorName, err)
-		}
-		return ctrl.Result{}, nil
-	}
-
-	err = r.updateCOStatus(ReasonSyncing, "", "Applying the Metal3 deployment")
+	err = r.updateCOStatus(ReasonSyncing, "syncing metal3 pod", "")
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Syncing state: %v", clusterOperatorName, err)
+		return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Progressing state: %v", clusterOperatorName, err)
 	}
 
-	// Proceed with creating a new Metal3 deployment
+	// Create metal3 deployment object based on current baremetal configuration
 	metal3Deployment := provisioning.NewMetal3Deployment(ComponentNamespace, &containerImages, &baremetalConfig.Spec)
-	expectedGeneration := resourcemerge.ExpectedDeploymentGeneration(metal3Deployment, r.Generations)
 
-	err = controllerutil.SetControllerReference(baremetalConfig, metal3Deployment, r.Scheme)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to set controllerReference on deployment: %v", err)
-	}
-
-	deployment, updated, err := resourceapply.ApplyDeployment(r.KubeClient.AppsV1(),
-		events.NewLoggingEventRecorder(ComponentName), metal3Deployment, expectedGeneration)
+	err = provisioning.ApplyNewOrUpdatedDeployment(r.KubeClient.AppsV1(), ComponentNamespace, metal3Deployment, r.Generations, ComponentName)
```

Comment:
```suggestion
	err = provisioning.ApplyDeployment(r.KubeClient.AppsV1(), ComponentNamespace, metal3Deployment, r.Generations, ComponentName)
```

Reply:
I was hoping that calling it ApplyNewOrUpdatedDeployment() will make it clear that sometimes we are just updating the deployment object with only portions that have changed since the previous reconcile run.

Reply:
:man_shrugging: I thought that was what apply ment - not a biggie, just looks overly wordy

Reply:
I like the short name too. I was trying to be unnecessarily "helpful" I guess.

Reply:
Not to be confused with this ApplyDeployment https://github.com/openshift/cluster-baremetal-operator/pull/65/files#diff-1575ce96065be1a97bee923445ae60115c8ce02b4a2736788012df8162407100R508.